Dockerfile.debug

```diff
@@ -0,0 +1,44 @@
+# syntax = docker/dockerfile:1
+
+ARG GO_VERSION="1.19"
+ARG RUNNER_IMAGE="alpine:3.16"
+
+FROM golang:${GO_VERSION}-alpine as builder
+
+WORKDIR /opt
+RUN apk add --no-cache make git gcc musl-dev openssl-dev linux-headers
+RUN CGO_ENABLED=0 go install -ldflags "-s -w -extldflags '-static'" -v github.com/go-delve/delve/cmd/dlv@latest
```

Comment:
Can we consolidate dockerfiles? Maybe just pass the `DOCKER_DEBUG` as an argument 

I'm also not sure you need an entrypoint, you can probably just override the command in the docker-compose-debug